core/capabilities.go

```diff
@@ -281,30 +276,111 @@ func (c *Capabilities) ToNetCapabilities() *net.Capabilities {
 	if c == nil {
 		return nil
 	}
-	return &net.Capabilities{Bitstring: c.bitstring, Mandatories: c.mandatories}
+	netCaps := &net.Capabilities{Bitstring: c.bitstring, Mandatories: c.mandatories, Capacities: make(map[uint32]uint32)}
```

Comment:
Makes sense to use mutex here as this looks like place of consumption of `.capacities` by upstream code.
```suggestion
	netCaps := &net.Capabilities{Bitstring: c.bitstring, Mandatories: c.mandatories, Capacities: make(map[uint32]uint32)}
	c.mutex.Lock()
	defer c.mutex.Unlock()
```

Reply:
Fixed